warplib/src/warp_memory/arrays.cairo

```diff
@@ -0,0 +1,91 @@
+use warplib::warp_memory::WarpMemory;
+use warplib::warp_memory::WarpMemoryTrait;
+use warplib::warp_memory::WarpMemoryImpl;
+use integer::u256_from_felt252;
+use integer::u128_try_from_felt252;
+use integer::Into;
+use option::OptionTrait;
+
+trait WarpMemoryArraysTrait {
+    /// Reads the pointer of an array. If it does not exist, it creates a new one
+    fn get_or_create_id(ref self: WarpMemory, location: felt252, size: felt252) -> felt252;
+
+    /// Given the length and the felt size of an element, allocates the space
+    /// in memory and returns a pointer to it
+    fn new_dynamic_array(ref self: WarpMemory, len: felt252, elem_width: felt252) -> felt252;
+
+    /// Given a pointer to a dynamic array, the index you want to access from it
+    /// and the size of the element in such array, returns a pointer to the element
+    fn index_dyn(ref self: WarpMemory, array_ptr: felt252, index: felt252, elem_width: felt252) -> felt252;
+
+    /// Given a pointer to a static array, the index you want to access from it,
+    /// the size of the elements in the array and the it's length,  returns a pointer
+    /// to the element
+    fn index_static(ref self: WarpMemory, array_ptr: felt252, index: felt252, elem_width: felt252, length: felt252) -> felt252;
+
+    /// Given the pointer to a dynamic array, it returns it's length
+    fn length_dyn(ref self: WarpMemory, array_ptr: felt252) -> felt252;
+}
+
+
+impl WarpMemoryArraysImpl of WarpMemoryArraysTrait {
+    fn get_or_create_id(ref self: WarpMemory, location: felt252, size: felt252) -> felt252{
+        // Get the id at the location
+        let id = self.read(location);
+        if id != 0 {
+            return id;
+        }
+
+        // If no id was found create a new one
+        let id = self.alloc(size);
+        self.unsafe_write(location, id);
+        id
+    }
```

Comment:
This is the place I was talking about which can improve naming of static/dynamic addresses. `id` and `location` are  unintuitive, especially `id`. But as we spoke, let's fix it in later PRs

Reply:
yes, I understand. Which name you think would be descriptive enough without being overly verbose?

Reply:
Maybe instead of `location` it could be `pointer_position` and instead of `id` it could be `array_location`? So the name of function could be `get_array_location_or_alloc_new`. WYT?